meta/autoid/autoid.go

```diff
@@ -165,21 +200,38 @@ func (alloc *allocator) Alloc(tableID int64) (int64, error) {
 			if err1 != nil {
 				return errors.Trace(err1)
 			}
-			newEnd, err1 = m.GenAutoTableID(alloc.dbID, tableID, step)
+			tmpStep := step
+			if alloc.isUnsigned {
+				tmpStep = int64(mathutil.MinUint64(math.MaxUint64-uint64(newBase), uint64(step)))
+			} else {
+				tmpStep = mathutil.MinInt64(math.MaxInt64-newBase, step)
+			}
+			newEnd, err1 = m.GenAutoTableID(alloc.dbID, tableID, tmpStep)
 			if err1 != nil {
 				return errors.Trace(err1)
 			}
 			return nil
 		})
 		metrics.AutoIDHistogram.WithLabelValues(metrics.TableAutoIDAlloc, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
 		if err != nil {
-			return 0, errors.Trace(err)
+			return 0, err
 		}
 		alloc.base, alloc.end = newBase, newEnd
 	}
+	log.Warning(uint64(alloc.base), uint64(alloc.end))
```

Comment:
log.Warning what ?

Reply:
Just for debugging, can be removed.

Reply:
Please check https://github.com/pingcap/tidb/pull/8181/commits/758fd314f1011612482f51e68c19b923c99126eb.